### PR TITLE
[web] Move filtering out cloud and tcp apps to the frontend

### PR DESCRIPTION
**Pull request by kimlisa**
#### Description
The logic to filter out `cloud and tcp` apps originated from here: https://github.com/gravitational/teleport/pull/13455/files#r896739044, which is causing issues with the true `totalCount` of apps which is required in the web UI to accurately relay to users how many total apps exists. And we also didn't provide an opt out of this filter, if we required all types of apps (required when listing apps with `searchAsRoles`)

Because the decision to hide unsupported apps from the web UI was not an official product/ux decision, I decided to move this filter logic to the frontend where if its a `cloud or tcp` app (determined by URI), we will simply not render the `launch` button.

This way, it also aligns with how `tsh apps ls` works, which lists every app you have access to.

Future improvements can be made to show a button that shows a dialog with instructions on how to access from CLI.

Snuck in a css change.

#### Screenshot
image

### CODE REVIEW

**Comment by zmb3** [comment]
How does this relate to @rudream's PR? Are you both solving the same problem?

**Comment by rudream** [comment]
> How does this relate to @rudream's PR? Are you both solving the same problem?

Yes this addresses the same problem. My PR was made as a temporary fix assuming we weren't decided yet on how we wanted to approach displaying unsupported apps in the UI. If this is merged, my PR can be closed.

**Review by avatus — APPROVED** [review]

File: web/packages/teleport/src/Apps/AppList/AppList.tsx [hunk 1]
```diff
@@ -158,7 +168,12 @@ function renderLaunchButtonCell({
   fqdn,
   clusterId,
   publicAddr,
+  uri,
 }: App) {
+  if (isCloudOrTcpApp(uri)) {
+    return <Cell></Cell>;
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply web/packages/teleport/src/Apps/AppList/AppList.tsx]: image

File: web/packages/teleport/src/Apps/AppList/AppList.tsx [hunk 2]
```diff
@@ -182,6 +197,10 @@ function renderLaunchButtonCell({
   return <Cell align="right">{$btn}</Cell>;
 }
 
+function isCloudOrTcpApp(uri: string) {
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply web/packages/teleport/src/Apps/AppList/AppList.tsx]: done, i also refactored the `making of addr` to the `App` type itself as well

**Review by zmb3 — APPROVED** [review]

**Review by rudream — APPROVED** [review]

**Pull request merged**